### PR TITLE
Adds support to run and compare benchmarks on CI.

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
## Description

This PR adds the support for running benchmarks in Promscale on CI. It also compares with the benchmark results of the master branch.

By default, benchmarks will not be active when a PR (or draft PR) is opened. A dev has to assign `action:benchmarks` label, which will automatically trigger the benchmark workflow.

In order to restart the benchmarks, just remove and reassign the label, and it will trigger the benchmark workflow again.

Note:
1. After the benchmark comparison with benchstat is done, the results will be pasted as a PR comment, tagging the author. Each PR comment with a result contains the code hash corresponding to which the benchmark was executed.2. 
2. The commenting on PR will happen only if the PR is opened from a branch in Promscale repo, not from your fork. This is due to some permission reason.

You can refer to https://github.com/timescale/promscale/pull/1505 to see the testing in verbose level.

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [ ] CHANGELOG entry for user-facing changes
- [ ] Updated the relevant documentation

### CODE REVIEW

**Comment by Harkishen-Singh** [comment]
Now, lets attach the `action:benchmarks` to trigger the `Benchmarks` workflow.

File: .github/workflows/benchmark.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        uses: actions/checkout@v3
+
+      - name: Set up Go ${{ env.golang-version }}
+        uses: actions/setup-go@v3.2.1
+        with:
+          go-version: ${{ env.golang-version }}
+        id: go
+
+      - name: Benchmark current PR
+        run: go test -bench=. -run=^$ -benchmem -cpu=1 ./... | tee current.txt
```
Comment (JamesGuthrie) [inline .github/workflows/benchmark.yaml:28]: The only concern I have is that with only 1 run, we may not get concrete enough results to be able to determine if there was a difference or not.

**Review by JamesGuthrie — APPROVED** [review]

File: .github/workflows/benchmark.yaml (hunk 1 above)
Comment (Harkishen-Singh) [reply .github/workflows/benchmark.yaml:28]: Do you mean, with `-cpu=1` or with `-benchtime=<some fixed iterations>`?

**Comment by Harkishen-Singh** [comment]
Closing this as just realized

> The commenting on PR will happen only if the PR is opened from a branch in Promscale repo, not from your fork. This is due to some permission reason.

File: .github/workflows/benchmark.yaml (hunk 1 above)
Comment (JamesGuthrie) [reply .github/workflows/benchmark.yaml:28]: I meant the `-count` parameter. Implicitly we're setting `-count=1`. I'm not sure that I understand it well yet, but I've had situations where count was low, and benchstat would basically say that it didn't have enough data to determine if the outcome was statistically significant or not.

**Pull request closed without merging**